### ISSUE TITLE
Autocorrect bug with Explicit Init rule

### ISSUE THREAD

**Issue by BB9z** [realm/SwiftLint#5289]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Steps to reproduce

```shell
$ swiftlint --version
0.53.0

$ echo "f { e in\n    A.init(e: e)\n}" >> z1.swift

$ cat z1.swift
f { e in
    A.init(e: e)
}

$ swiftlint z1.swift --autocorrect --enable-all-rules
Correcting Swift files at paths z1.swift
Correcting 'z1.swift' (1/1)
/Users/.../Documents/z1.swift:2:6 Corrected Explicit Init
Done correcting 1 file!

$ cat z1.swift
f { e inA(e: e)
}
```

The lint result should be

```swift
f { e in
    A(e: e)
}
```
